toplevel/coqloop.ml

```diff
@@ -371,8 +371,90 @@ let exit_on_error =
   declare_bool_option_and_ref ~depr:false ~name:"coqtop-exit-on-error" ~key:["Coqtop";"Exit";"On";"Error"]
     ~value:false
 
+(* XXX: This is duplicated with Vernacentries.show_proof , at some
+   point we should consolidate the code *)
+let show_proof_diff_to_pp pstate =
+  let p = Option.get pstate in
+  let sigma, env = Pfedit.get_proof_context p in
+  let pprf = Proof.partial_proof p in
+  Pp.prlist_with_sep Pp.fnl (Printer.pr_econstr_env env sigma) pprf
+
+let show_proof_diff_cmd ~state removed =
+  let open Vernac.State in
+  try
+    let n_pp = show_proof_diff_to_pp state.proof in
+    if true (*Proof_diffs.show_diffs ()*) then
+      let doc = state.doc in
+      let oproof = Stm.get_prev_proof ~doc (Stm.get_current_state ~doc) in
+      try
+        let o_pp = show_proof_diff_to_pp oproof in
+        let tokenize_string = Proof_diffs.tokenize_string in
+        let show_removed = Some removed in
+        Pp_diff.diff_pp_combined ~tokenize_string ?show_removed o_pp n_pp
+      with
+      | Pfedit.NoSuchGoal
+      | Option.IsNone -> n_pp
+      | Pp_diff.Diff_Failure msg -> begin
+          (* todo: print the unparsable string (if we know it) *)
+          Feedback.msg_warning Pp.(str ("Diff failure: " ^ msg) ++ cut()
+                                   ++ str "Showing results without diff highlighting" );
+          n_pp
+        end
+    else
+      n_pp
+  with
+  | Pfedit.NoSuchGoal
+  | Option.IsNone ->
+    CErrors.user_err (str "No goals to show.")
+
+let process_stm ~state stm =
```

Comment:
Please call this process_statement, since STM is already a thing

Reply:
fixed